lib/foreman_rh_cloud/engine.rb

```diff
@@ -210,6 +210,29 @@ def self.register_scheduled_task(task_class, cronline)
       )
     end
 
+    # Ideally this code belongs to an initializer. The problem is that Katello controllers are not initialized completely until after the end of the to_prepare blocks
+    # This means I can patch the controller only in the after_initialize block that is promised to run after the to_prepare
+    # initializer 'foreman_rh_cloud.allow_smart_proxy_actions', :before => :finisher_hook, :after => 'katello.register_plugin'  do |_app|
+    # end
+    config.after_initialize do
+      if defined?(Katello)
+        Katello::Api::V2::OrganizationsController.include Foreman::Controller::SmartProxyAuth
+        Katello::Api::V2::OrganizationsController.add_smart_proxy_filters(
+          :download_debug_certificate,
+          features: ForemanRhCloud.on_prem_smart_proxy_features
+        )
+        Katello::Api::V2::OrganizationsController.add_smart_proxy_filters(
+          :index,
+          features: ForemanRhCloud.on_prem_smart_proxy_features
+        )
+        Katello::Api::V2::RepositoriesController.include Foreman::Controller::SmartProxyAuth
```

Comment:
Perhaps visually separate the various controllers?
```suggestion

        Katello::Api::V2::RepositoriesController.include Foreman::Controller::SmartProxyAuth
```